File: Android/app/src/main/java/com/sjtu/runner/route/RouteInfo.kt
package com.sjtu.runner.route

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class RouteInfo(
    val id: String,
    val name: String,
    val filePath: String,
    val isDefault: Boolean = false,
    val pointCount: Int = 0,
    val createdAt: Long = System.currentTimeMillis()
) : Parcelable

File: IOS/SJTURunningMan/RouteInfo.swift
//
//  RouteInfo.swift
//  SJTURunningMan
//
//  Created by Jie Tang on 2026/7/8.
//

import Foundation

struct RouteInfo: Codable, Identifiable, Equatable {
    let id: String
    let name: String
    let filePath: String      // relative path in Documents/routes/, or "" for default
    let isDefault: Bool
    let pointCount: Int
    let createdAt: Date

    static func == (lhs: RouteInfo, rhs: RouteInfo) -> Bool {
        lhs.id == rhs.id
    }
}
